### ISSUE TITLE
Place namespaces without any specific Project annotation in the "Default" Project

### ISSUE THREAD

**Issue by leodotcloud** [rancher/rancher#16493]
Sample Workflow:

- Setup Rancher
- Create a kubernetes cluster
- Grab the kubeconfig and setup kubectl on my laptop
- Run any example yaml files or sample libraries or helm chart
- Go to the UI and try to find the newly created stuff, nothing is visible.

It would greatly improve User Experience if we can by default place all the newly created namespaces in the "Default" project in Rancher. Later if the user wants to (after understanding the concept of Project) can move these namespaces to a different Project.

Related:
https://github.com/rancher/rancher/issues/15474

**Comment by janeczku** [comment]
Related/DuplicateOf https://github.com/rancher/rancher/issues/15820

**Comment by dawidmalina** [comment]
I think it would be nice if we could also assign namespace to project by project name. For example by adding label `cattle.io/projectName=default` or `cattle.io/projectName=system` or `cattle.io/projectName=someOtherProject`.

**Comment by gabrielcalderon** [comment]
I'm running into this same issue, it's so annoying. I'm trying to automate some stuff using Rancher CLI and it's impossible.

**Comment by dawidmalina** [comment]
I agree this is annoying and looks like no chance for implementation (at least for 2.x) :(

**Comment by kalise** [comment]
very bad :(

**Comment by Symbianx** [comment]
A way to also define the default project on which the namespaces would be added to would be awesome as well!

**Comment by gabrielcalderon** [comment]
@Symbianx, Agree! I can't import any yaml's into Rancher UI because the namespaces get created with no project :-/ 

The only work around is to create the Namespace first by hand and then import the application yaml.

**Comment by Symbianx** [comment]
@gabrielcalderon Another workaround is specifying the project-id annotation & label in the namespace. You can see those if you `kubectl get namespace <existing-namespace> -o yaml` on an existing namespace.

**Comment by gabrielcalderon** [comment]
@Symbianx Still a pain when using other peoples yaml that already include a definition for a Namespace. We spent over an hour trying to figure out why our workloads were not showing up in Rancher's UI, but showed up in kubecrl. Turns out, the namespace had no project assigned....

**Comment by Ronsevet** [comment]
This is a huge blocker for actually enjoying Rancher UI abilities with a CI/CD pipeline

**Comment by strowi** [comment]
any news on this? this is especially annoying with gitlabs review-apps.

**Comment by sahina** [comment]
Any progress one this????

**Comment by thecrazyrussian** [comment]
Looks like this can be accomplished by using OPA to manually set the project by adding an annotation to the namespace using the  workaround in https://github.com/rancher/rancher/issues/21222.

Here is an example of using opa to mutate annotations: https://gist.github.com/tsandall/f328635433acc5beeb4cb9b36295ee89

**Comment by bruno-lopes** [comment]
What do you think about the following proposal?

https://github.com/rancher/rancher/issues/26825

**Comment by gabrielcalderon** [comment]
@bruno-lopes That still won't work for open source Helm charts that require a specific Namespace.

**Comment by ebauman** [comment]
You can use this helm chart to accomplish this goal: https://github.com/ebauman/rancher-project-mapper

See the README in the chart to understand how to configure.

**Comment by gaby** [comment]
3 years later, and still no solution :-)

**Comment by dev-ago** [comment]
~~3~~ 4 years later, and still no solution 😅

**Comment by szimonek** [comment]
6  years later, and still no solution :-D

**Comment by janeczku** [comment]
Oh my, how the time flies! 😆 ⌚
